Yoast/wordpress-seo
# Strip all HTML tags except for allowed tags from howto and faq

## Summary

This PR can be summarized in the following changelog entry:

* Allows a subset of HTML tags in FAQ answer, HowTo description and HowToStep description schema output: `<h1><h2><h3><h4><h5><h6><br><ol><ul><li><a><p><b><strong><i><em>`

## Relevant technical choices:
* I've decided to not create a function for the stripping because it's not unlikely that Google will allow different tags in different places.
* I've corrected some unit tests: when the user doesn't add a howto description, howto title or howto step description, the jsonText or jsonName attributes are present, but empty. Before my changes, those attributes were not present in the objects used for testing, which led to failing tests.
* I've moved `class-schema-how-to-double.php` to the correct folder.

## Test instructions

This PR can be tested by following these steps:

* Create new post with a HowTo block
* View the post.
* Go to the database, find the post in the wp_post table. Edit the post_content: include some tags from the above-allowed tags list, and some other tags (e.g. `<div>`) in the jsonDescription, jsonText and jsonName.
    * Reload the post on the frontend (the backend will show a broken block. No need to look there 😉).
    * The allowed tags shouldn't be stripped from the jsonDescription and jsonText. The other tags shouldn't.
    * All tags should be stripped from the jsonName.

* Create new post with an FAQ block
* View the post.
* Go to the database, find the post in the wp_post table. Edit the post_content: include some tags from the above-allowed tags list, and some other tags (e.g. `<div>`) in the jsonAnswer.
    * Reload the post on the frontend (the backend will show a broken block. No need to look there 😉).
    * The allowed tags shouldn't be stripped from the jsonAnswer. The other tags shouldn't.

N.B. It purely strips tags. No spaces are added.

## UI changes

## Documentation

## Quality assurance


Fixes #13113
